x/emissions/abci.go

```diff
@@ -143,8 +149,14 @@ func DistributeObserverRewards(
 	// Processing Step 2: Emit the observer emissions
 	keeper.EmitObserverEmissions(ctx, finalDistributionList)
 
-	// Processing Step 3: Delete all matured ballots and the ballot list
-	emissionsKeeper.GetObserverKeeper().ClearMaturedBallotsAndBallotList(ctx, maturityBlocks)
+	// Processing Step 3a: Delete all finalized ballots at `maturityBlocksForFinalizedBallots` height
+	// This step optionally deletes the `BallotListForHeight` if all ballots are finalized and deleted
```

Comment:
`// This step optionally deletes the `BallotListForHeight` if all ballots are finalized`?

Reply:
okay, this step does deletion as well.